js/ad_manager_vast.js

```diff
@@ -667,8 +727,28 @@ OO.Ads.manager(function(_, $) {
       this.embedCode = this.amc.currentEmbedCode;
       this.allAdInfo = movieMetadata.ads || pbMetadata.all_ads;
       this.movieMd = movieMetadata;
-      if (pbMetadata && pbMetadata.tagUrl) {
-        this.adURLOverride = pbMetadata.tagUrl;
+      if (pbMetadata) {
+        if (pbMetadata.tagUrl) {
+          this.adURLOverride = pbMetadata.tagUrl;
+        }
+
+        if (pbMetadata.vpaidTimeout) {
+          if (typeof pbMetadata.vpaidTimeout.iframe === "number") {
+            this.VPAID_AD_IFRAME_TIMEOUT = pbMetadata.vpaidTimeout.iframe * 1000;
+          }
+
+          if (typeof pbMetadata.vpaidTimeout.loaded === "number") {
+            this.VPAID_AD_LOADED_TIMEOUT = pbMetadata.vpaidTimeout.loaded * 1000;
+          }
+
+          if (typeof pbMetadata.vpaidTimeout.started === "number") {
+            this.VPAID_AD_STARTED_TIMEOUT = pbMetadata.vpaidTimeout.started * 1000;
+          }
+
+          if (typeof pbMetadata.vpaidTimeout.stopped === "number") {
```

Comment:
Two things: Is 0 valid --> Should we do if( timeoutVar && _.isNumber(timeoutVar))?

Are negative values valid?